src/pages/register/Register.tsx

```diff
@@ -0,0 +1,515 @@
+  MenuItem,
+  Button,
+  Stack
+} from '@mui/material';
+import { ArrowForward } from '@mui/icons-material';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { Background } from '../../assets/images';
+import { listProvinces, listDistricts, listWards } from '../../data/fake';
+import {
+  IDistrict,
+  IDistricts,
+  IProvince,
+  IProvinces,
+  IWard,
+  IWards
+} from '../../interfaces';
+const Wrapper = styled.div`
+  width: 100vw;
+  height: 100vh;
+  display: block;
+  position: relative;
+  background-color: #ffffff;
+  padding: 0;
+  overflow-x: hidden;
+`;
+const Container = styled.div`
+  width: 100%;
+  height: 100%;
+  padding: 0;
+  display: flex;
+`;
+const SideLeft = styled.div`
+  width: 50%;
+  height: 100%;
+  display: flex;
+  flex-direction: row;
+  align-items: flex-start;
+  background: url(${Background});
+  background-size: cover;
+`;
+const SideRight = styled.div`
+  width: 50%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+const Form = styled.form`
+  width: 600px;
+  margin: auto;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+const Header = styled.div`
+  width: 100%;
+  height: 42px;
+  margin-bottom: 15px;
+`;
+const InputComponent = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: flex-start;
+  margin-bottom: 15px;
+`;
+const Label = styled.label`
+  margin-bottom: 8px;
+`;
+const Note = styled.span`
+  color: red;
+`;
+const DialogActions = styled.div`
+  width: 420px;
+  display: flex;
+  justify-content: flex-end;
+`;
+
+interface IFormData {
+  card: string;
+  email: string;
+  password: string;
+  fullName: string;
+  birthday: string;
+  gender: string;
+  province: string;
+  district: string;
+  ward: string;
+}
+
+const schema = yup
+  .object({
+    email: yup.string().required('Email là bắt buộc').email().trim(),
+    password: yup
+      .string()
+      .min(8)
+      .required('Mật khẩu dài ít nhất 8 kí tự')
+      .trim(),
+    fullName: yup.string().required('Tên không được để trống').trim(),
+    birthday: yup.string().required().trim(),
+    card: yup
+      .string()
+      .required('CCCD là bắt buộc')
+      .matches(/^[0-9]+$/, 'CMND phải là dạng số ')
+      .matches(/^(\d{9}|\d{12})$/, 'CMND chỉ chứa 9 hoặc 12 số'),
+    gender: yup.string().required(),
+    province: yup.string().required().trim(),
+    district: yup.string().required().trim(),
+    ward: yup.string().required().trim()
+  })
+  .required();
+
+const Register = () => {
+  const navigate = useNavigate();
+  const [provinces, setProvinces] = useState<IProvinces>([]);
+  const [districts, setDistricts] = useState<IDistricts>([]);
+  const [wards, setWards] = useState<IWards>([]);
```

Comment:
```suggestion
  const [provinces, setProvinces] = useState<IProvince[]>([]);
  const [districts, setDistricts] = useState<IDistrict[]>([]);
  const [wards, setWards] = useState<IWard[]>([]);
```

Ko cần phải viết type cho arrays